pkg/operator/staticresourcecontroller/static_resource_controller.go

```diff
@@ -265,6 +278,28 @@ func (c *StaticResourceController) Sync(ctx context.Context, syncContext factory
 	errors := []error{}
 	var notFoundErrorsCount int
 	for _, conditionalManifest := range c.manifests {
+		ready, err := c.precondition(ctx, conditionalManifest.files)
```

Comment:
I would expect the precondition to be only run once and have the rest of the logic be handled by the conditional manifest methods `shouldCreateFn()` and `shouldDeleteFn()`

Reply:
No, definitely not!  The precondition is expected to change over time. 

The contract for `shouldCreateFn` evaluates if something should be created once and acts upon it. The precondition may change and once it does, you should proceed with the reconciliation loop. Giving you an example - the Cert Manager Operator blocks the reconciliation loop **until** you remove any existing (like IBM's) Operator. Once you do - it will go ahead and create everything for you.

Reply:
Ah, I see, I was unclear - by "run once" I meant it should not be run in a for-cycle but prior to it.

Reply:
aaaah... yeah, I was thinking about this. However, `StaticResourcesPreconditionFuncType` takes `files []string` as an argument and I'd need aggregate a list of files from all `[]conditionalManifests`. So it's bad this or another way, I'd say.

Do you have any advise, which one is better?

Reply:
I wouldn't expect the precondition to be dependent on a list of files. I would actually expect the terms of the precondition to be constant when you're creating the controller (e.g. "there is no other instance of a cert manager operator in the cluster" - no dependency on any files).

I see that in you're PR you're actually doing exactly that - https://github.com/openshift/cert-manager-operator/pull/29/files#diff-e11833beee2c6440e2fdea89051a509fb504d11388c3ea7bbfeb64ea8f93f772R24 - the string slice is ignored.

Reply:
Ok, agreed.

Reply:
LGTM